From bf939355f0be048ea1d16d8b40e78052c4e17565 Mon Sep 17 00:00:00 2001
From: Tom Rondeau <tom@trondeau.com>
Date: Tue, 24 Sep 2013 13:17:28 -0400
Subject: [PATCH 29/40] cmake: improve OOT projects find the installed GNU
 Radio.

1. Do not install FindGnuradio.cmake; everything is done through GnuradioConfig.cmake.

2. Fix a small bug with finding the PMT library if pkg-config is used.

3. Updated the documentation for how to use these tools in the OOT project.
---
 CMakeLists.txt                     |    3 +++
 cmake/Modules/GnuradioConfig.cmake |    2 +-
 docs/doxygen/other/main_page.dox   |   50 +++++++++++++++++++++++++-----------
 3 files changed, 39 insertions(+), 16 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf96d3e..57e6474 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -317,6 +317,9 @@ endif(NOT CMAKE_MODULES_DIR)
 
 # Install all other cmake files into same directory
 file(GLOB cmake_others "cmake/Modules/*.cmake")
+list(REMOVE_ITEM cmake_others
+    "${CMAKE_SOURCE_DIR}/cmake/Modules/FindGnuradio.cmake"
+)
 
 install(
   FILES ${cmake_configs} ${cmake_others}
diff --git a/cmake/Modules/GnuradioConfig.cmake b/cmake/Modules/GnuradioConfig.cmake
index 6410dfa..eae9a4a 100644
--- a/cmake/Modules/GnuradioConfig.cmake
+++ b/cmake/Modules/GnuradioConfig.cmake
@@ -120,4 +120,4 @@ GR_MODULE(UHD gnuradio-uhd gnuradio/uhd/api.h gnuradio-uhd)
 GR_MODULE(VOCODER gnuradio-vocoder gnuradio/vocoder/api.h gnuradio-vocoder)
 GR_MODULE(WAVELET gnuradio-wavelet gnuradio/wavelet/api.h gnuradio-wavelet)
 GR_MODULE(WXGUI gnuradio-wxgui gnuradio/wxgui/api.h gnuradio-wxgui)
-GR_MODULE(PMT gnuradio-pmt pmt/pmt.h gnuradio-pmt)
+GR_MODULE(PMT gnuradio-runtime pmt/pmt.h gnuradio-pmt)
diff --git a/docs/doxygen/other/main_page.dox b/docs/doxygen/other/main_page.dox
index d6cc269..f96492c 100644
--- a/docs/doxygen/other/main_page.dox
+++ b/docs/doxygen/other/main_page.dox
@@ -419,8 +419,8 @@ New as of 3.6.5.
 
 Using gr_modtool, each package comes with the ability to easily locate
 the gnuradio-runtime library using the 'find_package(GnuradioRuntime)'
-cmake command. This only locates that the library and include
-directories exist, which is enough for most simple projects.
+cmake command. This only locates the gnuradio-runtime library and
+include directory, which is enough for most simple projects.
 
 As projects become more complicated and start needing to rely on other
 GNU Radio components like gnuradio-blocks or gnuradio-filter, for
@@ -456,18 +456,38 @@ part of GNU Radio, the configuration will also fail.
 
 \subsection oot_config_path_page Install Path
 
-Cmake has to know where to find these configuration files. They are
-installed into $prefix/lib/cmake/gnuradio. If $prefix is '/usr' or
-'/usr/local', then everything should work fine. If the GNU Radio
-install $prefix is something else, then Cmake must be told where to
-find it. This can be done in two ways. If you are installing the
-out-of-tree module into the same $prefix, then you would be setting
-'-DCMAKE_INSTALL_PREFIX' on the configuration command line. This is
-enough to tell Cmake where to look for the configuration files.
-
-The other way to do it is to set the CMAKE_PREFIX_PATH environmental
-variable to $prefix. You can then install your component anywhere
-you'd like and it will be able to find and configure against the
-installed GNU Radio.
+Cmake has to know where to find either the package config files or the
+GnuradioConfig.cmake script. The package config files are located in
+$prefix/lib/pkgconfig while all of the Cmake scripts from GNU Radio
+are installed into $prefix/lib/cmake/gnuradio.
+
+If the installed GNU Radio $prefix is '/usr' or '/usr/local', then
+everything should work fine. If the GNU Radio install $prefix is
+something else, then Cmake must be told where to find it. This can be
+done in a few ways:
+
+1. If you are installing the out-of-tree module into the same $prefix,
+then you would be setting '-DCMAKE_INSTALL_PREFIX' on the
+configuration command line. This is enough to tell Cmake where to look
+for the configuration files.
+
+2. Cmake will try to find the package config (*.pc) files. If it can,
+these files will instruct Cmake where to look for the rest of the
+configuration options. If this is not set, it can be set as:
+
+\code
+    export PKG_CONFIG_PATH=$prefix/lib/pkgconfg:$PKG_CONFIG_PATH
+\endcode
+
+3. Set the CMAKE_PREFIX_PATH environmental variable to $prefix.
+
+\code
+    export CMAKE_PREFIX_PATH=$prefix:$CMAKE_PREFIX_PATH
+\endcode
+
+
+With method 1, you will be installing your OOT project into the same
+$prefix as GNU Radio. With methods 2 and 3, you can install your
+component anywhere you like (using -DCMAKE_INSTALL_PREFIX).
 
 */
-- 
1.7.10.4

